### PR TITLE
Feature entity mapping

### PR DESCRIPTION
**Pull request by StudentLgotnik**

### CODE REVIEW

File: src/main/java/org/meme/corp/database/entity/Event.java [hunk 1]
```diff
@@ -0,0 +1,24 @@
...
+import javax.persistence.Entity;
+import javax.persistence.Id;
+import javax.persistence.Table;
+
+@Entity
+@Table(name = "event")
+@NoArgsConstructor
+@AllArgsConstructor
+@Setter
+@Getter
```
Comment (spanasiuk) [inline src/main/java/org/meme/corp/database/entity/Event.java]: Replace with @ Data and @ Builder

File: pom.xml [hunk 2]
```diff
@@ -25,6 +29,30 @@
...
+            <groupId>org.postgresql</groupId>
+            <artifactId>postgresql</artifactId>
+            <version>${postgresql.version}</version>
+        </dependency>
+
+<!--        <dependency>-->
+<!--            <groupId>org.flywaydb</groupId>-->
+<!--            <artifactId>flyway-core</artifactId>-->
+<!--            <version>${flyway.core.version}</version>-->
+<!--        </dependency>-->
```
Comment (spanasiuk) [inline pom.xml]: remove if not needed

File: pom.xml [hunk 3]
```diff
@@ -83,6 +111,11 @@
                     </execution>
                 </executions>
             </plugin>
+<!--            <plugin>-->
+<!--                <groupId>org.flywaydb</groupId>-->
+<!--                <artifactId>flyway-maven-plugin</artifactId>-->
+<!--                <version>${flyway.plugin.version}</version>-->
+<!--            </plugin>-->
```
Comment (spanasiuk) [inline pom.xml]: remove if not needed

File: pom.xml [hunk 4]
```diff
@@ -16,6 +16,10 @@
         <slf4j.version>1.7.30</slf4j.version>
         <postgresql.version>42.2.19</postgresql.version>
         <junit.version>5.7.2</junit.version>
+        <hibernate.version>5.5.0.Final</hibernate.version>
+        <postgresql.version>42.2.18</postgresql.version>
+<!--        <flyway.core.version>7.5.1</flyway.core.version>-->
+<!--        <flyway.plugin.version>4.2.0</flyway.plugin.version>-->
```
Comment (spanasiuk) [inline pom.xml]: remove if not needed

File: docker-compose.yml [hunk 5]
```diff
@@ -0,0 +1,21 @@
...
+services:
+  postgresqldb:
+    image: postgres:latest
+    container_name: postgres
+    ports:
+      - 5432:5432
+    environment:
+      - POSTGRES_DB=expense_tracker_db
+      - POSTGRES_USER=postgres
+      - POSTGRES_PASSWORD=admin
```
Comment (spanasiuk) [inline docker-compose.yml:12]: we need another way of providing the credentials instead of hardcoding in the git repository

File: docker-compose.yml [hunk 6]
```diff
@@ -0,0 +1,21 @@
...
+    restart: unless-stopped
+  flyway:
+    image: flyway/flyway
+    command: -url=jdbc:postgresql://postgresqldb:5432/expense_tracker_db -user=postgres -password=admin -connectRetries=60 migrate
```
Comment (spanasiuk) [inline docker-compose.yml:16]: same

File: src/main/resources/META-INF/persistence.xml [hunk 7]
```diff
@@ -0,0 +1,19 @@
...
+        <description>Hibernate EntityManager Expense Tracker</description>
+        <class>org.meme.corp.database.entity.Event</class>
+        <exclude-unlisted-classes>true</exclude-unlisted-classes>
+        <properties>
+            <property name="hibernate.dialect" value="org.hibernate.dialect.PostgreSQL82Dialect"/>
+            <property name="hibernate.hbm2ddl.auto" value="validate"/>
+            <property name="javax.persistence.jdbc.driver" value="org.postgresql.Driver"/>
+            <property name="javax.persistence.jdbc.url" value="jdbc:postgresql://localhost:5432/expense_tracker_db"/>
+            <property name="javax.persistence.jdbc.user" value="postgres"/>
+            <property name="javax.persistence.jdbc.password" value="admin"/>
```
Comment (spanasiuk) [inline src/main/resources/META-INF/persistence.xml]: we need a way of providing this data another way (like https://stackoverflow.com/a/33011196/4797648)

**Review by spanasiuk — CHANGES_REQUESTED** [review]

File: src/main/resources/META-INF/persistence.xml (hunk 7 above)
Comment (iSan4eZ) [reply src/main/resources/META-INF/persistence.xml:16]: https://stackoverflow.com/a/33011196/4797648

**Review by iSan4eZ — CHANGES_REQUESTED** [review]

File: pom.xml (hunk 2 above)
Comment (StudentLgotnik) [reply pom.xml]: I will remove it in the final commit if not needed

File: pom.xml (hunk 3 above)
Comment (StudentLgotnik) [reply pom.xml]: I will remove it in the final commit if not needed

File: src/main/java/org/meme/corp/database/entity/Event.java (hunk 1 above)
Comment (StudentLgotnik) [reply src/main/java/org/meme/corp/database/entity/Event.java]: Ok

File: src/main/resources/META-INF/persistence.xml (hunk 7 above)
Comment (StudentLgotnik) [reply src/main/resources/META-INF/persistence.xml]: Ok, I will try it

File: docker-compose.yml (hunk 5 above)
Comment (StudentLgotnik) [reply docker-compose.yml:12]: It is only for local usages instead of using a real database.

File: docker-compose.yml (hunk 6 above)
Comment (StudentLgotnik) [reply docker-compose.yml:16]: same

Comment (spanasiuk) [reply docker-compose.yml:16]: it's possible to put flyway credentials in the environment variables (https://flywaydb.org/documentation/configuration/parameters/). And for local use, you can create flyway.properties file and add it to gitignore (https://www.baeldung.com/database-migrations-with-flyway#3-external-configuration-file)

File: pom.xml [hunk 8]
```diff
@@ -25,39 +29,68 @@
...
         <plugins>
-            <plugin>
-                <groupId>org.apache.maven.plugins</groupId>
-                <artifactId>maven-compiler-plugin</artifactId>
-                <configuration>
-                    <compilerArgs>
-                        --enable-preview
-                    </compilerArgs>
-                </configuration>
-            </plugin>
```
Comment (spanasiuk) [inline pom.xml]: Why did you remove this?

File: src/main/java/org/meme/corp/database/entity/PersonClient.java [hunk 9]
```diff
@@ -0,0 +1,29 @@
...
+@Data
+@Builder
+public class PersonClient {
+
+    @Id
+    @GeneratedValue(strategy=GenerationType.IDENTITY)
+    private long id;
+
+    @Column(name = "client_name")
+    private String clientName;
```
Comment (spanasiuk) [inline src/main/java/org/meme/corp/database/entity/PersonClient.java]: Shouldn't it be the same composite key as in the chat?

File: pom.xml (hunk 8 above)
Comment (StudentLgotnik) [reply pom.xml]: Why not?

Comment (spanasiuk) [reply pom.xml]: because multiline strings won't work at least - those were released in Java 15

Comment (StudentLgotnik) [reply pom.xml]: Maybe multiline strings won't work because we use Java 14?

Comment (spanasiuk) [reply pom.xml]: But it will work with preview features enabled

Comment (StudentLgotnik) [reply pom.xml]: Check this https://app.circleci.com/pipelines/github/MemeCorp/expense-tracker-bot/25/workflows/288fa1e8-acec-487f-8597-984908491b79/jobs/27

File: pom.xml (hunk 2 above)
Comment (StudentLgotnik) [reply pom.xml]: Done

File: pom.xml (hunk 3 above)
Comment (StudentLgotnik) [reply pom.xml]: Done

File: pom.xml (hunk 4 above)
Comment (StudentLgotnik) [reply pom.xml]: Done

**Review by spanasiuk — APPROVED** [review]

File: pom.xml (hunk 8 above)
Comment (spanasiuk) [reply pom.xml]: resolved
